example/bench/recipe.cpp

```diff
@@ -0,0 +1,54 @@
+#include <random>
+
+#include <benchmark_cell.hpp>
+#include <common_types.hpp>
+#include <time_sequence.hpp>
+
+#include "recipe.hpp"
+
+cell_size_type bench_recipe::num_cells() const {
+    return params_.num_cells;
+}
+
+arb::util::unique_any bench_recipe::get_cell_description(cell_gid_type gid) const {
+    using RNG = std::mt19937_64;
+    arb::benchmark_cell cell;
+    cell.realtime_ratio = params_.cell.realtime_ratio;
+    cell.time_sequence = arb::poisson_time_seq<RNG>(RNG(gid), 0, 1e-3*params_.cell.spike_freq_hz);
+    return std::move(cell);
+}
+
+arb::cell_kind bench_recipe::get_cell_kind(arb::cell_gid_type gid) const {
+    return arb::cell_kind::benchmark;
+}
+
+std::vector<arb::cell_connection> bench_recipe::connections_on(cell_gid_type gid) const {
+    const auto n = params_.network.fan_in;
+    std::vector<arb::cell_connection> cons;
+    cons.reserve(n);
+    using RNG = std::mt19937_64;
+    RNG rng(gid);
+    // issue: invalid gid for source in a connection
+    std::uniform_int_distribution<cell_gid_type> dist(0, params_.num_cells-2);
+    for (unsigned i=0; i<n; ++i) {
+        cell_gid_type src = dist(rng);
+        if (src>=gid) ++src;
```

Comment:
I don't understand what is happening here. Why are we increment src when it's `>= gid`? If it was just `gid`, I'd assume we don't want ourselves to be the source, and so we make it "the next cell", but this ... I don't get. And what if we go over the max gid? What if the src is now outside of the range?

Reply:
I have added detailed comments that explain what is goind on here.
We draw values in the range [0, num_cells-2], so by adding one we get uniform values in the range [0, num_cells-1], hence we never get a source with gid greater than or equal to num_cells.